lib/spy.js

```diff
@@ -13,6 +13,165 @@ module.exports = function (chai, _) {
+    };
+    object[methodName] = method;
+
+    return method;
+  };
+
+  /**
+   * # Sandbox.object (function)
+   *
+   * Creates an object with spied methods if first argument is a String.
+   * Wraps passed object's methods with spies if first argument is an Object.
+   * In case, if first parameter is an object and second is omit wraps
+   * all enumerable methods of passed object.
+   *
+   *      var spy = chai.spy.sandbox();
+   *      var object = spy.object('Array', [ 'push', 'pop' ]);
+   *      var array = spy.object([], [ 'push', 'pop' ]);
+   *
+   * @param {String|Object} name object or object name
+   * @param {String[]} [methods] method names
+   * @returns object with spied methods
+   * @api public
+   */
+
+  Sandbox.prototype.object = function (name, methods) {
+    var defs = {};
+    var sandbox = this;
+    var object;
+
+    if (name && typeof name === 'object') {
+      object = name;
+      methods = methods || Object.keys(object);
+      methods.forEach(function (methodName) {
+        var method = object[methodName];
+
+        if (typeof method === 'function' && !method.__spy) {
+          sandbox.on(object, methodName);
+        }
+      })
+    } else {
+      object = methods.reduce(function (object, methodName) {
+        object[methodName] = chai.spy(name + '.' + methodName);
+        return object;
+      }, {});
+    }
+
+    return object;
+  };
+
+  /**
+   * # Sandbox.restore (function)
+   *
+   * Restores previously wrapped object's method.
+   * Restores all spied objects of a sandbox if called without parameters.
+   *
+   *      var spy = chai.spy.sandbox();
+   *      var object = spy.on(Array, 'isArray');
+   *      spy.restore(Array, 'isArray'); // or spy.restore();
+   *
+   * @param {Object} [object]
+   * @param {String} [methodName] method name
+   * @api public
+   */
+
+  Sandbox.prototype.restore = function (object, methodName) {
+    var hasFilter = Boolean(object && methodName);
+    var sandbox = this;
+
+    Object.keys(this[STATE_KEY]).some(function (spyId) {
+      var spy = sandbox[STATE_KEY][spyId];
+      var tracked = spy.__spy.tracked;
+      var isObjectSpied = !object || object === tracked.object;
+      var isMethodSpied = !methodName || methodName === tracked.methodName;
+
+      delete sandbox[STATE_KEY][spyId];
+
+      if (!isObjectSpied && !isMethodSpied) {
+        return false;
+      }
+
+      sandbox.restoreTrackedObject(spy);
+
+      if (hasFilter) {
+        return true;
+      }
+    });
+  };
+
+  /**
+   * # Sandbox.restoreTrackedObject (function)
+   *
+   * Restores tracked object's method
+   *
+   *      var spy = chai.spy.sandbox();
+   *      var isArray = spy.on(Array, 'isArray');
+   *      spy.restoreTrackedObject(isArray);
+   *
+   * @param {Spy} spy
+   * @api private
+   */
+
+  Sandbox.prototype.restoreTrackedObject = function (spy) {
+    var tracked = spy.__spy.tracked;
+
+    if (!tracked) {
+      return;
+    }
```

Comment:
I don't like this silent failure path. We should throw an error here, telling the user they cannot restore a non-tracked spy.